app/use_cases/calculate_scores.rb

```diff
@@ -131,4 +135,13 @@ def round_result(value, score)
 
     value.round(score[:round_to_decimals])
   end
+
+  def preprocess_value(numeric_value, qid, score)
+    return numeric_value unless score.key?(:preprocessing) && score[:preprocessing].key?(qid)
+
+    preprocessing_step = score[:preprocessing][qid]
+    numeric_value *= str_or_num_to_num(preprocessing_step[:multiply_with]) if preprocessing_step.key?(:multiply_with)
```

Comment:
it would be even nicer if you could create like a hash of function / key or something in the known preprocessing steps constant (something like `{name: :multiply_with, function: '*='}` or however you would do that in ruby. In that case you only have to keep one list, while now you have to keep this one consistent with the other list, right?

Reply:
I see what you mean. How can you get a reference to a method in ruby though?

Reply:
Found it.

Reply:
Ok this actually cleanes up the code a whole lot.

Reply:
Thanks for this. The code is awesome now.

Reply:
Fixed.